### ISSUE TITLE
I found a way to install ubuntu 20.04 (focal)

### ISSUE THREAD

**Issue by qwerty65432** [meefik/linuxdeploy#1260]
If you want please comment.
I will tell you if you need

**Comment by qwerty65432** [comment]
isn't there anyone that need run ubuntu 20.04 on linux deploy?

**Comment by falkor2k15** [comment]
@qwerty65432 I managed to upgrade from Bionic (18.04) all the way up to 21.10!

Firstly, update /etc/apt/sources.list to include bionic-updates main restricted entry then run sudo apt install update-manager-core

1. sudo apt update && sudo apt dist-upgrade
2. sudo do-release-upgrade

Repeat the above 2 steps for each incremental upgrade.

When you get to LTS release update this file and change to prompt=normal
sudo nano /etc/update-manager/release-upgrades

When you get to 21.04 instead of running sudo do-release-upgrade run sudo do-release-upgrade -d (for development version)

You got a better way?

**Comment by qwerty65432** [comment]
idk, but i think its a bit better.
it's better to install focal instead of upgrade to focal.
good thing about linux deploy, you can change almost every thing on installing and configuring linux distros:
to install focal after you install linux deploy run linux deploy and then exit.(run programm makes it create ENV
directory and files)
on your phone go to /data/user/ru.meefik.linuxdeploy/files/include,bootstrap/

**Comment by falkor2k15** [comment]
Was it just this line you changed?

 --suite="focal"
     Version of Linux distribution, supported versions "precise", "trusty", "xenial", "focal" and "bionic".

How about directly install hirsute? I see it's listed here:
http://archive.ubuntu.com/ubuntu/dists/

It is just a case of changing that one word above, from focal to hirsute?

**Comment by qwerty65432** [comment]
really?
i dont have any problems when im using it

**Comment by falkor2k15** [comment]
Not sure what you mean by "brain" being connected to my character name? 
Focal still seems to work besides the "font" error. Going to try hirsute next - let me know if you get impish working?
BTW, i also trying to find out how to get LXQT working, and I'm interested in what file manager you use on Android?

**Comment by qwerty65432** [comment]
I mean exactly "how to improve "intelligence"" that I think it depends on time and neurons.
If you want to know about it, I think we can talk about it.
I use mixplorer (silver version).
I think mixplorer is best android file manager and it have many great tools, it have code editor, text editor, db editor, archive manger an many many tools, it's beautiful (also you can find and install many beautiful themes)
You can download silver version on
https://www.farsroid.com/mixplorer-android/

**Comment by qwerty65432** [comment]
I interestd in which keyboard(s) you use.?

**Comment by falkor2k15** [comment]
Not really sure about how intelligence works (or AI for that matter), but I'm guessing it's based on time, effort, memory, open-mindedness, holistic understanding (trees amongst the forest) and seeing how everything is connected.

So far my best file explorers have been total commander and solid explorer. Sentio file manager would have been the best, but it cannot browse the system root.

I am looking for some ways to bring a desktop experience to Android, but so far I only found Taskbar 6.0 to help on that front. Linux Deploy is proving quite limited too in other ways based on lack of graphics acceleration, so I'm trying to find a compromise between using Android and Linux Deploy.

All my keyboards are UK (wired and bluetooth), but it seems the default layout with VNC Servers is US, so copy and paste + arrow keys don't work for me inside any Linux Deploy distro, unfortunately.  

Any suggestions?

**Comment by falkor2k15** [comment]
I found out that the VNC copy and paste problem is mostly down to the Terminal app and not the keyboard layout:
Termux: CTRL+ALT+V
Linux Terminal: CTRL+SHIFT+V

**Comment by qwerty65432** [comment]
hi falkor, are u steel alive?

**Comment by falkor2k15** [comment]
Yeah, not ready to give up yet - always learning new things. I haven't really compared linux deploy vs. proot yet, but I am guessing they will be the same speed since they both use the same Android kernel, and the proot is just a non-root exploit of the normal (root) way of running Linux through Linux Deploy. Unfortunately, they both suffer through lack of graphics accleration as well as running inside a container. A virtual machine would have been better, but Android security simply doesn't allow a hypervisor to run except on less than a handful of phones/chipsets (I know only of the ASUS Zenfone 2 with its x86 kernel) if only there was a list available - bearing in mind that the claim of there being even a handful of such phones might be false. It's unfortunate there's so many restrictions on phones in terms of the bootloader compared to PCs since the hardware/price nowadays are comparable between different form-factors, based on specs.

**Comment by falkor2k15** [comment]
linux deploy upgrade work, but for some reason proot upgrades fail with "fatal error", so there must be some, other, fundamental difference between them.

**Comment by qwerty65432** [comment]
thank u

**Comment by raymondcc2tm** [comment]
> @qwerty65432 I managed to upgrade from Bionic (18.04) all the way up to 21.10!
> 
> Firstly, update /etc/apt/sources.list to include bionic-updates main restricted entry then run sudo apt install update-manager-core
> 
> 1. sudo apt update && sudo apt upgrade
> 2. sudo do-release-upgrade
> 
> Repeat the above 2 steps for each incremental upgrade.
> 
> When you get to LTS release update this file and change to prompt=normal sudo nano /etc/update-manager/release-upgrades
> 
> When you get to 21.04 instead of running sudo do-release-upgrade run sudo do-release-upgrade -d (for development version)
> 
> You got a better way?

"sudo do-release-upgrade -d" for 1st time, and "sudo do-release-upgrade" thereafter.

**Comment by sinhpn92** [comment]
Hello @falkor2k15  @qwerty65432 ,
Did you install ubuntu oracular? I have got issue `base-files_13.3ubuntu6_arm64.deb requires the zstdcat command, which is not available`.
